src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -662,6 +663,7 @@ private void ExecuteIntrinsicTask(ItemBucket bucket)
         /// </summary>
         private async Task<WorkUnitResult> InitializeAndExecuteTask(TaskLoggingContext taskLoggingContext, ItemBucket bucket, IDictionary<string, string> taskIdentityParameters, TaskHost taskHost, TaskExecutionMode howToExecuteTask)
         {
+            using var assemblyLoadsTracker = AssemblyLoadsTracker.StartTracking(taskLoggingContext, AssemblyLoadingContext.TaskRun);
```

Comment:
Why not be specific about `AppDomain.CurrentDomain` here to make it explicit it's part of the tracking?